fedimint-server/src/consensus/mod.rs

```diff
@@ -40,6 +43,8 @@ pub struct FedimintConsensus {
     pub db: Database,
     /// API for accessing state
     pub client_cfg_hash: sha256::Hash,
+    /// track the latest contribution by peer
+    pub latest_contribution_by_peer: Arc<RwLock<LatestContributionByPeer>>,
```

Comment:
Will this be exposed via the API/UI? Or what is this data for? If not: why the `RwLock`? I hope we don't actually expect parallel writes? If not, and it is only for the UI, then maybe a [tokio watch channel](https://docs.rs/tokio/latest/tokio/sync/watch/fn.channel.html) or similar wold be appropriate?

Reply:
This is exposed via the api to flag peers as lagging. Did not choose the RwLock for this, we do not have parallel writes though. I was thinking to use a database entry to track this to ensure consistency across shutdowns